lib/matplotlib/figure.py

```diff
@@ -862,19 +870,85 @@ def figimage(self, X, xo=0, yo=0, alpha=None, norm=None, cmap=None,
         self.stale = True
         return im
 
+    def set_size(self, w, h=None, units=None, forward=True):
```

Comment:
Now that we're introducing a new API, let's decide whether we want `set_size((w, h))` or `set_size(w, h)` but don't support both.  (I have a mild preference for `set_size((w, h))`, as that fits the setter model better (`figure.set(size=(w, h))`)).

Reply:
I don’t think it necessarily is a good idea to make it inconsistent with the existing  API.  Why do we have to enforce a tuple?

Reply:
I agree that this should be decided. Unfortunately there are good reasons for both.

`set_size((w, h))` is more difficult to type and read. It feels a bit artificial, also given that `ax.set_xlim(left, right)` has a similar signature and I wouldn't vote for changing that.

OTOH, `set_size(size)` would allow to trivially support `size=(6, 4, "cm")` throughout the whole library and in all figure-creating functions. If we just have `set_size(w, h, unit)` we might need some explicit conversion of the tuple.

All over, I have a mild preference for `set_size(w, h)`.

Reply:
Actually, `set_xlim((left, right))` also works...

Reply:
Yeah the `boo(x, y)` = `boo((x, y))` API is quite prevalent.  I agree it can make parsing the args a bit of a nuisance, though to me the basic problem w/ `plot` is the fact we allow, `plot(x1, y1, x2, y2)`.

Reply:
If it was just that. We also have `plot(x1, y1, fmt1, x2, y2, fmt2)` and even `plot(y1, fmt1, y2, fmt2)`. Welcome to the Matlab way of thinking. But we are getting off-topic.